src/Sulu/Bundle/EventLogBundle/Infrastructure/Doctrine/Repository/EventRecordRepository.php

```diff
@@ -62,13 +67,41 @@ public function createForDomainEvent(DomainEvent $domainEvent): EventRecordInter
         return $eventRecord;
     }
 
-    public function add(EventRecordInterface $eventRecord): void
+    public function addAndCommit(EventRecordInterface $eventRecord): void
     {
-        $this->entityManager->persist($eventRecord);
-    }
+        // use persister to insert only given entity instead of flushing all managed entities via the entity manager
+        // this prevents flushing unrelated changes and allows to call this method in a postFlush event-listener
 
-    public function commit(): void
-    {
-        $this->entityManager->flush();
+        $userId = null;
+        if ($eventRecord->getUser()) {
+            $userId = $eventRecord->getUser()->getId();
+        }
+
+        $queryBuilder = $this->entityManager->getConnection()->createQueryBuilder();
+        $queryBuilder
+            ->insert('el_event_records')
+            ->values(
+                [
```

Comment:
think i would chain `setValue()` calls here, but probably just a matter of taste 🙂